### PR TITLE
test(fw): Unit tests for post state verification errors

### PR DESCRIPTION
**Pull request by winsvega**
## 🗒️ Description
Adds filling framework unit tests for post state verification, of accounts including storage, nonce, code & balance mismatch. Ensure that the framework post state mismatches are occuring correctly.

## 🔗 Related Issues
N/A.

## ✅ Checklist
- [x] All: Set appropriate labels for the changes.
- [x] All: Considered squashing commits to improve commit history.
- [x] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [x] All: Considered updating the online docs in the [./docs/](../docs/) directory.

### CODE REVIEW

File: src/ethereum_test_tools/tests/test_filler_expect.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            ),
+        }
+        fixture.clear()
+    except Storage.KeyValueMismatch as e:
+        wasError = True
+        assert e.want == 2
+        assert e.got == 1
+        assert e.key == 1
+        assert e.address == "0x1000000000000000000000000000000000000000"
+    assert wasError
```
Comment (danceratopz) [inline src/ethereum_test_tools/tests/test_filler_expect.py:79]: pytest has a helper [`pytest.raises`](https://docs.pytest.org/en/latest/how-to/assert.html#assertions-about-expected-exceptions) to check that correct exceptions get raised which provides more informative info than manually checking:
```suggestion
    e_info: pytest.ExceptionInfo
    with pytest.raises(Storage.KeyValueMismatch) as e_info:
        fixture = {
            f"000/my_chain_id_test/{fork}": fill_test(
                t8n=t8n,
                test_spec=state_test,
                fork=fork,
                spec=None,
            ),
        }
    assert e_info.value.want == 2
    assert e_info.value.got == 1
    assert e_info.value.key == 1
    assert e_info.value.address == "0x1000000000000000000000000000000000000000"
    assert "incorrect value in address" in str(e_info.value)
```

**Review by danceratopz — DISMISSED** [review]
Thanks! Would love more like this!

Below are a few pytest specific suggestions.

File: src/ethereum_test_tools/tests/test_filler_expect.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+@pytest.mark.parametrize(
+    "fork,enable_hive",
+    [
+        (Istanbul, False),
+        (London, False),
+        (Merge, True),
+        (Shanghai, True),
+    ],
+)
+def test_post_expect_section(fork: Fork, enable_hive: bool):
```
[the remark this answers is not in the document]
Comment (danceratopz) [reply src/ethereum_test_tools/tests/test_filler_expect.py:26]: (If you'd like to test other post sections, I'd probably split them into other tests, perhaps with some helper functions).

File: src/ethereum_test_tools/tests/test_filling/test_expect.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+post = {}
+
+
+def run_post_state_mismatch_test(
+    pre: Mapping[Any, Any], post: Mapping[Any, Any], exception: Any
+) -> Optional[pytest.ExceptionInfo]:
+    """
+    Performs test filling and post state verification.
+    """
+    fork = Shanghai
```
Comment (danceratopz) [inline src/ethereum_test_tools/tests/test_filling/test_expect.py:30]: Is there a good alternative to hard-coding the fork here? :thinking: 

How about using `ethereum.test_forks.get_deployed_forks()[-1]`?

File: src/ethereum_test_tools/tests/test_filling/test_expect.py [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+        fixture_format=fixture_format,
+    )
+
+    if exception:
+        e_info: pytest.ExceptionInfo
+        with pytest.raises(exception) as e_info:
+            state_test.generate(t8n=t8n, fork=fork)
+        return e_info
+    else:
+        return None
```
Comment (danceratopz) [inline src/ethereum_test_tools/tests/test_filling/test_expect.py:48]: Imo, we still have to call `state_test.generate()` here, in order to check that we **don't get an exception**.
```suggestion
    else:
        state_test.generate(t8n=t8n, fork=fork)
        return None
```

**Review by danceratopz — COMMENTED** [review]
A couple of things occurred to me whilst working on #380, see comments below.

**Comment by danceratopz** [comment]
I tried to express the test's intent a bit better by removing the split of exception checking across the helper function and the test. Now we have the setup code in the `state_test` fixture and the actual test is in the test. See #393.

**Review by spencer-tb — APPROVED** [review]
LGTM!

**Pull request merged**